frontend/src/components/table/schedule/sections/shifts-section/shifts-section.tsx

```diff
@@ -20,14 +20,27 @@ export function ShiftsSectionComponent({
 }: ShiftsSectionOptions) {
   const logic = new ShiftsInfoLogic(DataRowHelper.dataRowsAsValueDict<Shift>(data, true));
 
+  const [isOpened, setIsOpened] = useState(false);
+
+  const modal = (
+    <AddWorkerModal
+      isOpened={isOpened}
+      setIsOpened={setIsOpened}
+      submit={submit}
+      workerType={workerType}
+    />
+  );
+
   function addDataRow(newRow: DataRow) {
     onSectionUpdated([...logic.sectionData, newRow]);
   }
 
   function onAddButtonClicked() {
+    setIsOpened(true);
+  }
+
+  function submit(name, time) {
```

Comment:
@Qwebeck Nie wiem co zrobić z wielkością etatu. Czy jest już zapisywana gdzieś w stanie?

Reply:
Na razie nie jest. Będziemy to handlować już po refaktoringu